hw/ip/rom_ctrl/util/scramble_image.py

```diff
@@ -157,14 +157,18 @@ class Scrambler:
     subst_perm_rounds = 2
     num_rounds_half = 3
 
-    def __init__(self, nonce: int, key: int, rom_base: int, rom_size_words: int,
+    def __init__(self, nonce: int, nonce_width: int,
+                 key: int, key_width: int,
+                 rom_base: int, rom_size_words: int,
                  hash_file: IO[str]):
-        assert 0 <= nonce < (1 << 64)
+        assert 0 <= nonce < (1 << 128)
         assert 0 <= key < (1 << 128)
```

Comment:
Don't these need tweaking so the assertions look like `0 <= nonce < (1 << nonce_width)` ? And probably we should throw in a couple of checks that `nonce_width` and `key_width` are non-negative as well?

Reply:
Makes sense, yes. Fixed now.